api/osb/check_platform_id_plugin.go

```diff
@@ -90,7 +91,17 @@ func (p *checkPlatformIDPlugin) assertPlatformID(req *web.Request, next web.Hand
 		}
 		return nil, util.HandleStorageError(err, string(types.ServiceInstanceType))
 	}
-	instance := object.(*types.ServiceInstance)
+
+	instance, ok := object.(*types.ServiceInstance)
```

Comment:
can this happen? you just loaded the instance from DB so it has to be a service instance, and if the instance was not found in DB the flow does not get here